JOSESwift/Sources/CompressionImplementation/Compressor.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  Compressor.swift
+//  JOSESwift
+//
+//  Created by Florian Häser on 13.02.19.
+//
+
+import Foundation
+
+protocol CompressorProtocol {
+    /// Compresses data using the `CompressionAlgorithm`.
+    ///
+    /// - Parameter decompressed data.
+    /// - Returns: The compressed data.
+    func compress(data: Data) -> Data
+    /// Decompresses data using the `CompressionAlgorithm`.
+    ///
+    /// - Parameter compressed data.
+    /// - Returns: The decompressed data.
+    func decompress(data: Data) -> Data
+}
+
+/// A `Compressor` that takes the data and passes it back without doing any compression or decompression.
+/// Used for having the JWE implementation more readable.
+struct NoneCompressor: CompressorProtocol {
+    func compress(data: Data) -> Data {
+        return data
+    }
+    func decompress(data: Data) -> Data {
+        return data
+    }
+}
+
+struct CompressorFactory {
+    /// Select the appropriate `Compressor` for a given `CompressionAlgorithm. Defaults to the `NoneCompressor`.
+    ///
+    /// - Parameter the `CompressionAlgorithm` for selecting the appropriate compressor.
+    /// - Returns: The appropriate compressor
+    static func makeCompressor(algorithm: CompressionAlgorithm?) -> CompressorProtocol {
```

Comment:
As mentioned above, I'd suggest explicitly only returning the none-compressor when we get the `.NONE` algorithm and throwing (or returning an optional) otherwise. Then we can reject unknown values.

Reply:
👍 I go with the throwing.